flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -0,0 +1,168 @@
+                    actualSchema = schema.getTypes().get(0);
+                    nullable = true;
+                } else if (schema.getTypes().size() == 1) {
+                    actualSchema = schema.getTypes().get(0);
+                    nullable = false;
+                } else {
+                    // use Kryo for serialization
+                    return new AtomicDataType(
+                            new TypeInformationRawType<>(false, Types.GENERIC(Object.class)));
+                }
+                DataType converted = convertToDataType(actualSchema);
+                return nullable ? converted.nullable() : converted;
+            case MAP:
+                return DataTypes.MAP(
+                                DataTypes.STRING().notNull(),
+                                convertToDataType(schema.getValueType()))
+                        .notNull();
+            case STRING:
+                DataType logicalDataType = handleLogicalTypeSchema(schema);
+                if (logicalDataType != null) {
+                    return logicalDataType;
+                }
+                return DataTypes.STRING().notNull();
+            case ENUM:
+                return DataTypes.STRING().notNull();
+            case FIXED:
+                // logical decimal type
+                logicalDataType = handleLogicalTypeSchema(schema);
+                if (logicalDataType != null) {
+                    return logicalDataType;
+                }
+                // convert fixed size binary data to primitive byte arrays
+                return DataTypes.VARBINARY(schema.getFixedSize()).notNull();
+            case BYTES:
+                // logical decimal type
+                logicalDataType = handleLogicalTypeSchema(schema);
+                if (logicalDataType != null) {
+                    return logicalDataType;
+                }
+                return DataTypes.BYTES().notNull();
+            case INT:
+                // logical date and time type
+                logicalDataType = handleLogicalTypeSchema(schema);
+                if (logicalDataType != null) {
+                    return logicalDataType;
+                }
+                return DataTypes.INT().notNull();
+            case LONG:
+                // logical timestamp type
+                logicalDataType = handleLogicalTypeSchema(schema);
+                if (logicalDataType != null) {
+                    return logicalDataType;
+                }
+                return DataTypes.BIGINT().notNull();
+            case FLOAT:
+                return DataTypes.FLOAT().notNull();
+            case DOUBLE:
+                return DataTypes.DOUBLE().notNull();
+            case BOOLEAN:
+                return DataTypes.BOOLEAN().notNull();
+            case NULL:
+                return DataTypes.NULL();
+        }
+        throw new IllegalArgumentException("Unsupported Avro type '" + schema.getType() + "'.");
```

Comment:
let's logs what the supported types are here

Reply:
Fixed.